### ISSUE TITLE
Matrix: Remove delivered messages and use room history

### ISSUE THREAD

**Issue by ulope** [raiden-network/raiden#2383]
## Problem Definition

We currently use `Delivered` messages on the Matrix transport to track whether messages arrived ath the destination.
Since Matrix is a reliable transport we don't need those. This would also mean to get rid of part of the re-try mechanism and instead rely on the room history for past messages.

## Tasks

- [ ] Investigate if removing the Delivered messages and using the room history works for our requirements
- [ ] Adjust re-try implementation to only rely on `Processed` messages 
- [ ] Make sure the room history is taken into account upon login (this should already be the case)
- [ ] Add tests

**Comment by andrevmatos** [comment]
About 3rd point, not yet the case: the rooms are created on the initial sync, which is also when the messages history is fetched, i.e.: before the room's listeners are added. We could do the initial sync again after the rooms were created and listeners added, but that would be not optimal, or ensure caching these messages and calling the listeners after the listeners were added.
Also, there's some limit for the size of the initial history retrieved, we would need to check to ensure we retrieved all non-readed messages, which arises another point: we need a way to know which messages were already processed, to avoid duplicate processing.
Finally, Raiden currently depends on some retry logic: for the messages which are cleaned on `Processed`, they contain a `nonce` for ordering on a per-channel basis, and sometimes the processing of these messages may fail initially, but would eventually succeed (e.g. a message which depended on the partner be on a given block, which it didn't retrieve yet). We would need to ensure some retry-logic on receiver, as well as a way to tell the sender the message was successfuly processed and they may proceed.
That said, I think this approach is the most efficient, and we should pursue it.
